### PR TITLE
test(e2e): run on a free port to avoid clashing with dev server

### PR DESCRIPTION
**Pull request by omarkohl**
The E2E web server and baseURL were pinned to localhost:3000, the same
port as `make dev`, causing intermittent clashes. Pick a free port
instead and freeze it into E2E_PORT so every Playwright worker (which
re-evaluates this config) agrees with the server. This also keeps
concurrent E2E runs in different jj workspaces from colliding.

Co-Authored-By: Claude Opus 4.8 <noreply@anthropic.com>

### CODE REVIEW

**Comment by omarkohl** [comment]
### Changes since last push

<details>
<summary><code>playwright.config.ts</code> (+25, -2)</summary>

```diff
index f810a23328..a3dde39729 100644
--- a/playwright.config.ts
+++ b/playwright.config.ts
@@ -7,6 +7,17 @@
  * short-lived helper process because this config module has no top-level await.
  * Override with E2E_PORT to point the tests at an already-running server.
  */
+function parsePort(raw: string): number {
+  if (!/^\d+$/.test(raw.trim())) {
+    throw new Error(`E2E_PORT must be a positive integer, got "${raw}"`);
+  }
+  const port = Number(raw.trim());
+  if (!Number.isInteger(port) || port < 1 || port > 65535) {
+    throw new Error(`E2E_PORT must be between 1 and 65535, got "${raw}"`);
+  }
+  return port;
+}
+
 function findFreePort(): number {
   const result = spawnSync(
     process.execPath,
@@ -14,8 +25,20 @@
       "-e",
       'const s=require("net").createServer();s.listen(0,()=>{process.stdout.write(String(s.address().port));s.close();});',
     ],
-    { encoding: "utf8" }
+    { encoding: "utf8", timeout: 5000 }
   );
+  if (result.error) {
+    throw new Error(`Failed to spawn port-finder: ${result.error.message}`, {
+      cause: result.error,
+    });
+  }
+  if (result.status !== 0) {
+    throw new Error(
+      `Port-finder exited with status ${result.status}` +
+        (result.stderr ? `\nstderr: ${result.stderr}` : "") +
+        (result.stdout ? `\nstdout: ${result.stdout}` : "")
+    );
+  }
   const port = Number.parseInt(result.stdout.trim(), 10);
   if (!Number.isInteger(port) || port <= 0) {
     throw new Error(`Failed to determine a free port (got "${result.stdout}")`);
@@ -28,7 +51,7 @@
 // the web server in agreement. E2E_PORT can also be set by hand to target an
 // already-running server.
 const port = process.env.E2E_PORT
-  ? Number.parseInt(process.env.E2E_PORT, 10)
+  ? parsePort(process.env.E2E_PORT)
   : findFreePort();
 process.env.E2E_PORT = String(port);
 const baseURL = `http://localhost:${port}`;
```

</details>

---
<sub>View the diff on [GitHub](https://github.com/LWCW-Europe/schellingboard/compare/5e6875a2715e118d1eebbc21683f70d8ba175bd8..9feea3849b54d46e27f5bf521a05f46b55968d1e) (may include unrelated changes due to rebases since GitHub does not currently implement `git range-diff`).
View the diff locally (will only work if you fetched the older commit at some point):
`git range-diff main 5e6875a 9feea38`
`jj interdiff -f 5e6875a -t 9feea38`
</sub>

**Comment by omarkohl** [comment]
### Changes since last push

<details open>
<summary><code>playwright.config.ts</code> (+5, -4)</summary>

```diff
index a3dde39729..9ffbce4843 100644
--- a/playwright.config.ts
+++ b/playwright.config.ts
@@ -2,10 +2,11 @@
 import { spawnSync } from "node:child_process";
 
 /**
- * Pick a free TCP port so E2E runs never collide with `make dev` (port 3000) or
- * with an E2E run in a different jj workspace. Done synchronously via a
- * short-lived helper process because this config module has no top-level await.
- * Override with E2E_PORT to point the tests at an already-running server.
+ * Pick a free TCP port so E2E runs never collide with `make dev` (port 3000)
+ * or with an E2E run in a different directory (workspace). Done synchronously
+ * via a short-lived helper process because this config module has no top-level
+ * await.  Override with E2E_PORT to point the tests at an already-running
+ * server.
  */
 function parsePort(raw: string): number {
   if (!/^\d+$/.test(raw.trim())) {
```

</details>

---
<sub>View the diff on [GitHub](https://github.com/LWCW-Europe/schellingboard/compare/9feea3849b54d46e27f5bf521a05f46b55968d1e..41bb811fc8110732d4ff5cc5e35949320771af3b) (may include unrelated changes due to rebases since GitHub does not currently implement `git range-diff`).
View the diff locally (will only work if you fetched the older commit at some point):
`git range-diff main 9feea38 41bb811`
`jj interdiff -f 9feea38 -t 41bb811`
</sub>

**Comment by omarkohl** [comment]
I'll ignore the last review comment by Qodo because it adds too much complexity for a big corner case (race condition between finding a free port and actually using it, meaning another process _could_ grab it in between). We are only talking about running tests, so the small risk is fine.

**Comment by omarkohl** [comment]
### Changes since last push

**No code changes** (likely just a rebase).

---
<sub>View the diff on [GitHub](https://github.com/LWCW-Europe/schellingboard/compare/41bb811fc8110732d4ff5cc5e35949320771af3b..c27f971b69bc83004325494381b89998d680ac06) (may include unrelated changes due to rebases since GitHub does not currently implement `git range-diff`).
View the diff locally (will only work if you fetched the older commit at some point):
`git range-diff main 41bb811 c27f971`
`jj interdiff -f 41bb811 -t c27f971`
</sub>

**Comment by omarkohl** [comment]
Merging myself because nobody is available to review

**Comment by omarkohl** [comment]
### Changes since last push

**No code changes** (likely just a rebase).

---
<sub>View the diff on [GitHub](https://github.com/LWCW-Europe/schellingboard/compare/c27f971b69bc83004325494381b89998d680ac06..ee5f6a9bcb4f13b565104cb4c8cb8a3c51f16b47) (may include unrelated changes due to rebases since GitHub does not currently implement `git range-diff`).
View the diff locally (will only work if you fetched the older commit at some point):
`git range-diff main c27f971 ee5f6a9`
`jj interdiff -f c27f971 -t ee5f6a9`
</sub>
